packages/medusa/src/utils/validators/is-type.ts

```diff
@@ -117,3 +119,57 @@ export function IsType(types: any[], validationOptions?: ValidationOptions) {
     })
   }
 }
+
+export function IsTypeJSONSchemaConverter(meta, options: Partial<IOptions>) {
```

Comment:
suggestion: I think it would be worth typing the expected types we can accept. wdyt?
```ts
type Constraint =
  | StringConstructor
  | BooleanConstructor
  | NumberConstructor
  | DateConstructor
  | Constraint[]

export function IsTypeJSONSchemaConverter(
  meta: { constraints: [Constraint[]] },
  options: Partial<IOptions>
) {
  // ...
}

export function inferType(typed: Constraint, options: Partial<IOptions>) { 
      // ... 
      if ("name" in typed && typed.name == "Object") {
        console.log(typed)
        return { type: "string" }
      }
      return {
        $ref:
          options.refPointerPrefix + (("name" in typed && typed.name) || ""),
      }
}

Reply:
This was a tad more involved than expected. IsType supports any type of Class, not just primitives. I did my best to at least improve from using `any`. See 622f848e5e59932277b1bcc4cbe66defcf3b836d